### ISSUE TITLE
Dark theme issues

### ISSUE THREAD

**Issue by Symbai** [NaBian/HandyControl#1017]
1. WPF's GridSplitter is not themed
![image](https://user-images.githubusercontent.com/14368203/137639543-453e4887-0b16-4875-a675-f4d5e1b1ab3b.png)
2. Selected/Active `TabControl` header foreground color is probably too dark. Its kinda hard to read. I think it would be better if its get the light blue color background as in bottom right (Title4) but as foreground
![image](https://user-images.githubusercontent.com/14368203/137639613-94945d02-9546-4aca-b8cd-8c46c77660a2.png)

Nuget version: 3.2.0 but you can also repro in master branch.

**Comment by NaBian** [comment]
@ghost1372 what do you think?

**Comment by Yopler** [comment]
If I propose my point of vue : 
I have the same feeling concerning the dark mode, color are too saturated (not sure that the right word 😅). The most obvious exemple is to read the code example in the demo : 
![image](https://user-images.githubusercontent.com/45144437/137736905-57a38c7b-1d9b-4dd7-bd49-0717249202b3.png)

In design, if you have a dark mode, color are useally less saturated (here again not sure of the word), like pastel. The goal is to have less agressiv colors for the user. 
Here an extreme exemple : 
![image](https://user-images.githubusercontent.com/45144437/137737952-a0b80cbc-2c2b-4463-81e1-912bb9a15767.png)
![image](https://user-images.githubusercontent.com/45144437/137737982-46301b43-a622-470c-9832-dbb41170335e.png)

**Comment by Yopler** [comment]
@NaBian do you have a design system for Handy Control ?

**Comment by ghost1372** [comment]
hmm....
how about this?
![Untitled](https://user-images.githubusercontent.com/9213496/137738546-74eb9e77-4dda-4128-bed8-a15f6864ff67.png)

or this
![Untitled2](https://user-images.githubusercontent.com/9213496/137738577-b7940278-8eb9-4fd5-8e4b-be62c911d1a0.png)

**Comment by ghost1372** [comment]
> @NaBian do you have a design system for Handy Control ?

look here
https://ant.design/components/tabs/

**Comment by Yopler** [comment]
Nice, You use ant design ! For Dark Mode, look https://ant.design/docs/spec/dark#Design-Principles.
You can try the level 8 of the color ! like the blue-8 : #65b7f3

Of course, you don't need to have the same color as Ant Design. But in that case, I think you need to have a Graphic Charter. 😄😉

**Comment by NaBian** [comment]
> If I propose my point of vue : I have the same feeling concerning the dark mode, color are too saturated (not sure that the right word 😅). The most obvious exemple is to read the code example in the demo : ![image](https://user-images.githubusercontent.com/45144437/137736905-57a38c7b-1d9b-4dd7-bd49-0717249202b3.png)
> 
> In design, if you have a dark mode, color are useally less saturated (here again not sure of the word), like pastel. The goal is to have less agressiv colors for the user. Here an extreme exemple : ![image](https://user-images.githubusercontent.com/45144437/137737952-a0b80cbc-2c2b-4463-81e1-912bb9a15767.png) ![image](https://user-images.githubusercontent.com/45144437/137737982-46301b43-a622-470c-9832-dbb41170335e.png)

update latest code, and you will see:
![image](https://user-images.githubusercontent.com/17383395/137752614-c4df1c53-1564-431c-bdf6-963590976761.png)

**Comment by ghost1372** [comment]
@NaBian can we change PrimaryBrush (Dark theme) for all controls? or we should only change for tabcontrol?

**Comment by NaBian** [comment]
for all controls, of course, while we should use blue-6 not blue-8
![1634568898(1)](https://user-images.githubusercontent.com/17383395/137756030-d3333ec1-7c06-4689-83a9-d823e65f7250.jpg)

**Comment by ghost1372** [comment]
> for all controls, of course, while we should use blue-6 not blue-8 ![1634568898(1)](https://user-images.githubusercontent.com/17383395/137756030-d3333ec1-7c06-4689-83a9-d823e65f7250.jpg)
They do not differ much😕
![Untitled](https://user-images.githubusercontent.com/9213496/137757378-8ca35422-12a2-410f-8850-4e5d577699ad.png)

**Comment by NaBian** [comment]
we don't have designers. let's do what they say 😆 .

**Comment by Symbai** [comment]
> hmm.... how about this? 
> 

Much better thank you. This was also a possible solution I thought of since the selected tab is underlined. I also thought about reporting the watermark of textboxes is hard to read. But I figured out it is also on light/white mode hard to read. In my opinion it could be the same color as the disabled textbox (disabled textbox is still different because the of the different border color).

**Comment by Yopler** [comment]
> update latest code, and you will see: ![image](https://user-images.githubusercontent.com/17383395/137752614-c4df1c53-1564-431c-bdf6-963590976761.png)

Oh Beautiful 🤩 
But the fact is that you have only change the color of the Code (https://github.com/HandyOrg/HandyControl/tree/master/src/Shared/HandyControlDemo_Shared/Resources/xshd) But not for the whole dark mode !

**Comment by Yopler** [comment]
> for all controls, of course, while we should use blue-6 not blue-8 ![1634568898(1)](https://user-images.githubusercontent.com/17383395/137756030-d3333ec1-7c06-4689-83a9-d823e65f7250.jpg)

Here folowing rule for contrast in design. It is here call for web but it can apply to software because this rules apply for any computer screen. Here your contrast is  4.38 . That's a little low 😅

> Web Content Accessibility Guidelines (WCAG) : 
> Regarding colors, the standard defines two levels of contrast ratio: AA (minimum contrast) and AAA (enhanced contrast).
> The level AA requires a contrast ratio of at least 4.5:1 for normal text and 3:1 for large text (at least 18pt) or bold text.
> The level AAA requires a contrast ratio of at least 7:1 for normal text and 4.5:1 for large text or bold text.

**Comment by Yopler** [comment]
I was using a [color contrast checker](https://coolors.co/contrast-checker) to test the colors blue-6 and blue-8 : 
Here the blue-8 : 
![image](https://user-images.githubusercontent.com/45144437/137867752-948d4f01-e2fc-4431-9628-08231413d992.png)

and here the blue-6 : 
![image](https://user-images.githubusercontent.com/45144437/137867847-19c8eae7-11e2-4800-be14-1a6ff616020d.png)

my point here, is that color used in color.xaml and colorDark.xaml are the same :

- PrimaryColor : #326cf3
- Danger color : #db3340
- Warning color : #e9af20
- Info Color : #e9af20
- SuccessCOlor : #2db84d

Background color like for button can stay the same but text color should change to follow contrast rule. Here for Tab you use the PrimaryColor and not the PrimaryTextColor.

**Comment by ghost1372** [comment]
level 8 color is good for tabs but not good for datagrid

level 8
![image](https://user-images.githubusercontent.com/9213496/137942414-7c2ff0ab-7b9a-4850-9649-001d5eaeba75.png)

But level 7 looks good

level 7

![image](https://user-images.githubusercontent.com/9213496/137942691-42905658-03fc-4e25-b7b0-8d5db05185cd.png)

**Comment by Yopler** [comment]
> level 8 color is good for tabs but not good for datagrid
> 
> level 8
> ![image](https://user-images.githubusercontent.com/9213496/137942414-7c2ff0ab-7b9a-4850-9649-001d5eaeba75.png)
> 
> But level 7 looks good
> 
> level 7
> 
> ![image](https://user-images.githubusercontent.com/9213496/137942691-42905658-03fc-4e25-b7b0-8d5db05185cd.png)
> 

Yes , that what I was explaining. The color contrast is is about **Text** on **background** ! 
The fact here, is that the Tab is Text and you use a background color. 
In your datagrid , the constrast is wrong because your blue-8 is your background (was text color / foreground in Tab) and your text is White

**Comment by Yopler** [comment]
When I got a little of time, I can try to make a figma for a graphic charter. It is not my profession, i have already done it sometimes for personnal project and for fun. I can try 🤔🤷‍♂️
